lib/gui/app.js

```diff
@@ -298,11 +298,18 @@ app.controller('HeaderController', function(SelectionStateModel, OSOpenExternalS
 
 });
 
-app.controller('StateController', function($state) {
+app.controller('StateController', function($rootScope, $state) {
 
   /**
    * @param {string} state - state page
    */
   this.is = $state.is;
 
+  /**
+   * @param {function} func - function to pass $rootScope to
+   */
+  this.onStateChange = (func) => {
```

Comment:
If this function is called multiple times, then multiple event listeners will be registered for this event.